RELEASE.md

```diff
@@ -2,22 +2,59 @@
 
 [![hackmd-github-sync-badge](https://hackmd.io/sVOAyv6LTwiQllQUctxP1w/badge)](https://hackmd.io/sVOAyv6LTwiQllQUctxP1w)
 
-## Resolve Issues
+## Versioning Scheme
+Flyte components follow [SemVer conventions](https://semver.org/). In a nutshell, that means every component has a release version that follows this format: `MAJOR.MINOR.PATCH`. This includes the aggregated milestone release described below.
+
+Flyte hasn't reached `v1.0.0` version yet. This version indicates a stable public API. After releasing `v1.0.0` any backward incompatible changes require a `MAJOR` version bump. The community and the team, however, strives to maintain backward compatibility with the current public APIs as much as possible and gracefully deprecating APIs when absolutely required.
+
+Until Flyte reaches `v1.0.0`, we are maintaining the following definitions of versions `MAJOR.MINOR.PATCHbX`:
+
+1. **MAJOR** is pinned to `0`.
+1. **MINOR** is bumped on one of two occasions:
+    1. A backward incompatible change or,
```

Comment:
i again want to be careful, this would imply no one will every upgrade minor versions!!
Its not a backwards incompatible change, but we can say that it may have alteration in behavior?